### PR TITLE
Add blog article about gRPC probes beta

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Placeholder PR for the blog post for gRPC

/sig node
KEP: https://github.com/kubernetes/enhancements/issues/2727

### CODE REVIEW

**Comment by sftim** [comment]
/retitle [WIP] Add blog article about gRPC probes beta

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+---
+layout: blog
+title: "Try the new gRPC probes in 1.24: now in beta."
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:3]:
```suggestion
title: "Try the new gRPC probes in 1.24: now in beta"
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+date: 2022-04-15
+slug: grpc-probes-now-in-beta
+---
+
+With Kubernetes 1.24 the gRPC probes functionality entered beta and is available by default.
+Now you can configure startup, liveness, and readiness probes for your gRPC app
+without exposing http endpoint or using an extra executable. All built-in to the Kubernetes.
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:10]:
```suggestion
without exposing any HTTP endpoint, nor do you need an executable. Kubernetes can natively
connect to your your workload via gRPC and query its status.
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+## Some history
+
+Kubernetes allows to control the container lifecycle from inside the app
+by checking http response, running executable, or simply connecting to a tcp port.
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:15]:
```suggestion
It's useful to let the system managing your workload check that the app is healthy,
has started OK, and whether the app considers itself good to accept traffic.
Before the gRPC support was added, Kubernetes already allowed you to check
for health based on running an executable from inside the container image,
by making an HTTP request, or by checking whether a TCP connection succeeded.
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+Those checks are often enough for the most of apps. For gRPC apps, it is easy
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:17]:
```suggestion
For most apps, those checks are enough. If your provides a gRPC endpoint for a health
(or readiness) check, it is easy
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
+---
+layout: blog
+title: "Try the new gRPC probes in 1.24: now in beta."
+date: 2022-04-15
+slug: grpc-probes-now-in-beta
+---
+
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:7]: Mention an author (see other articles for an example).

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+## Some history
+
+Kubernetes allows to control the container lifecycle from inside the app
+by checking http response, running executable, or simply connecting to a tcp port.
+
+Those checks are often enough for the most of apps. For gRPC apps, it is easy
+to repurpose the `exec` probe to use it for gRPC health checking. This solution
+was described in the blog post "[Health checking gRPC servers on Kubernetes](https://kubernetes.io/blog/2018/10/01/health-checking-grpc-servers-on-kubernetes/)"
+and used for a long time. Commonly used tool to enable this was created on [Aug 21, 2018](https://github.com/grpc-ecosystem/grpc-health-probe/commit/2df4478982e95c9a57d5fe3f555667f4365c025d) with
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:20]:
```suggestion
to repurpose the `exec` probe to use it for gRPC health checking.
In the blog article [Health checking gRPC servers on Kubernetes](/blog/2018/10/01/health-checking-grpc-servers-on-kubernetes/), Ahmet Alp Balkan described how you can do that—a
mechanism that still works today.

There is a commonly used tool to enable this that was [created](https://github.com/grpc-ecosystem/grpc-health-probe/commit/2df4478982e95c9a57d5fe3f555667f4365c025d)
on August 21, 2018, and with
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+the first release at [Sep 19, 2018](https://github.com/grpc-ecosystem/grpc-health-probe/releases/tag/v0.1.0-alpha.1).
+
+This approach for gRPC apps health checking is very popular. There are [3,626 Dockerfiles](https://github.com/search?l=Dockerfile&q=grpc_health_probe&type=code)
+with the `grpc_health_probe` and [6,621 yaml](https://github.com/search?l=YAML&q=grpc_health_probe&type=Code) files that are discovered with the
+basic search on GitHub (at the moment of writing). This is good indication of the tool popularity
+and the need to support this natively.
+
+With the Kubernetes 1.23, native support for gRPC health probing was introduced as alpha feature.
+The feature is not at beta and enabled by default.
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:29]:
```suggestion
Kubernetes v1.23 introduced an alpha-quality implementation of native support for
querying a workload status using gRPC.
Because it was an alpha feature, this was disabled by default for the v1.23 release.
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+## Using the feature
+
+The gRPC health checking is [easy to use](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe).
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:33]: You can reword this in two ways:
- express _your opinion_ that it is easy to use (opinions are OK in blog articles)
- use neutral language

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    --no-enable-autoupgrade \
+    --release-channel=rapid \
+    --cluster-version=1.23
+```
+
+You will also need to configure `kubectl` to access the cluster:
+
+``` bsh
+$ gcloud container clusters get-credentials test-grpc
+```
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:75]: - we should explain how to get a v1.24 cluster, not a v1.23 cluster
- we shouldn't require or imply that the user uses Google Cloud

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+
+### Trying the feature out
+
+Lets create the pod to test how gRPC probes work. The `agnhost` image has
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:79]: What's an `agnhost`? I imagine readers will ask that exact question.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+$ kubectl describe test-grpc
+
+...
+Conditions:
+  Type              Status
+  Initialized       True
+  Ready             True
+  ContainersReady   True
+  PodScheduled      True
+...
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:119]: This isn't a valid shell script. Consider following how the docs do it: [separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output).

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+Now let's change the health checking endpoint status to NOT_SERVING.
+In order to call the http port of the Pod, let's create a port forward:
+
+``` bsh
+$ kubectl port-forward test-grpc 8080:8080
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:126]: Don't include a prompt; it makes the code not be valid shell.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+You can `curl` to call the command...
+
+``` bsh
+$ curl http://localhost:8080/make-not-serving
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:132]: Don't include a prompt; it makes the code not be valid shell.
Also, the highlight for POSIX shell is `shell` not `bsh`.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+  Ready             False
+  ContainersReady   False
+  PodScheduled      True
+
+...
+
+  Warning  Unhealthy  2s (x6 over 42s)  kubelet            Readiness probe failed: service unhealthy (responded with "NOT_SERVING")
+```
+
+Once it is switched back, in ~1s the port will get back to ready status:
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:152]:
```suggestion
Once it is switched back, in about one second the port will get back to ready status:
```
Did you mean “Pod” instead of “port”?

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+
+Conditions:
+  Type              Status
+  Initialized       True
+  Ready             True
+  ContainersReady   True
+  PodScheduled      True
+```
+
+Using gRPC health probing on Kubernetes is easy and straightforward. Follow the [documentation to learn more](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe)
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:168]:
```suggestion
This new built-in gRPC health probing on Kubernetes makes implementing ahealth-check via gRPC
much easier than the older approach that relied on using a separate `exec` probe. Read through
the official
[documentation](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe)
to learn more.
```

**Review by sftim — DISMISSED** [review]
Thanks for this. Here's some partial feedback; I recommend incorporating as much of it as makes sense, and then we can review the updated PR.

Please also target the _main_ branch for this (as the blog article is not part of the release itself).

**Comment by sftim** [comment]
@SergeyKanzhelev because this is a blog article, the PR should target the _main_ branch. Could you switch the base branch over?

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev because this is a blog article, the PR should target the _main_ branch. Could you switch the base branch over?

Thank you for the review! With all your contributions, can I include you as an author?

**Comment by valaparthvi** [comment]
Hey @SergeyKanzhelev! I'm from the Comms team, and I wanted to check on the status of this blog. Reviews for feature blogs are supposed to start next week. Anything we can help you with?

**Comment by SergeyKanzhelev** [comment]
> Hey @SergeyKanzhelev! I'm from the Comms team, and I wanted to check on the status of this blog. Reviews for feature blogs are supposed to happen this week. Anything we can help you with?

All feedback from @sftim was addressed, will find somebody for tech aspects, but it should be ok.

**Comment by SergeyKanzhelev** [comment]
/assign @mrunalp 

for technical lgtm

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+
+You will also need to configure `kubectl` to access the cluster:
+
+``` bsh
+gcloud container clusters get-credentials test-grpc
+```
+
+### Trying the feature out
+
+Lets create the pod to test how gRPC probes work. For this test we will use the `agnhost` image.
```
Comment (mrunalp) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:95]: nit: Let's

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+    - containerPort: 8080
+    readinessProbe:
+      grpc:
+        port: 5000
+```
+
+If the file called `test.yaml`, you can create the pod and check it's status.
+The pod will be in ready state as indicated by the snippet of the output.
+
+``` bsh
```
Comment (mrunalp) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:125]: Is this supposed to be `bash`?

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 18]
```diff
@@ -0,0 +1,203 @@
...
+for those.
+
+## Try it for yourself
+
+### Getting cluster
+
+You can try this feature today. To try the feature you can spin up the kubernetes cluster
+yourself with the `GRPCContainerProbe` feature gate enabled, there are many [tools available](/docs/tasks/tools/).
+
+Alternatively, at the moment of writing, you can spin up the cluster on GKE for a quick test.
```
Comment (mrunalp) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:72]: Can we say that one needs to bring up a cluster on 1.24 and then for e.g. this is how you would do it on GKE. Also, mention that it should work for kind or any other 1.24 based kubernetes clusters.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 19]
```diff
@@ -0,0 +1,203 @@
...
+Conditions:
+  Type              Status
+  Initialized       True
+  Ready             True
+  ContainersReady   True
+  PodScheduled      True
+```
+
+This new built-in gRPC health probing on Kubernetes makes implementing a health-check via gRPC
+much easier than the older approach that relied on using a separate `exec` probe. Read through
```
Comment (mrunalp) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:194]: what about `easier and reduces overhead compared..`

**Comment by mrunalp** [comment]
Left some nits/suggestions but looks fine overall.

/lgtm

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md (hunk 18 above)
Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md]: Added a paragraph

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md (hunk 16 above)
Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md]: applied

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md (hunk 17 above)
Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md]: changed to ```shell for consistency with other docs

**Comment by SergeyKanzhelev** [comment]
Thank you @mrunalp!

@sftim please let me know if you have more comments. Also what date should I set as a publishing date now? (currently `date: 2022-04-15` as a placeholder)

**Comment by sftim** [comment]
The release comms team would assign a publication date, usually. I don't know what date is assigned for this one.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+the official
+[documentation](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe)
+to learn more and provide feedback before the feature will be promoted to GA.
+
+# Summary
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:204]:
```suggestion
## Summary
```
We don't use first-level headings within Markdown.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 21]
```diff
@@ -0,0 +1,208 @@
...
+```
+
+Once it is switched back, in about one second the Pod will get back to ready status:
+
+``` bsh
+curl http://localhost:8080/make-serving
+kubectl describe test-grpc
+```
+
+And output indicates that service went back to ready:
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:187]:
```suggestion
The output indicates that the Pod went back to being `Ready`:
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 22]
```diff
@@ -0,0 +1,208 @@
...
+You can `curl` to call the command...
+
+```shell
+curl http://localhost:8080/make-not-serving
+```
+
+... and in a few seconds the port status will switch to not ready.
+
+```shell
+kubectl describe test-grpc
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:162]: Did you mean:
```suggestion
kubectl describe pod test-grpc
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 23]
```diff
@@ -0,0 +1,208 @@
...
+This is a k8s maintained image with that can be used for all sorts of workload testing.
+For example, it has a useful [grpc-health-checking](https://github.com/kubernetes/kubernetes/blob/b2c5bd2a278288b5ef19e25bf7413ecb872577a4/test/images/agnhost/README.md#grpc-health-checking) module
+that exposes two ports - one is serving health checking service,
+another - http port to react on commands `make-serving` and `make-not-serving`.
+
+Here is an example pod definition. It starts the `grpc-health-checking` module,
+exposes ports `5000` and `8080`, and configures gRPC readiness probe:
+
+``` yaml
+apiVersion: v1
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:110]:
```suggestion
---
```

I tend to start YAML examples with `---`. It's not required but visually it helps people spot that they're reading YAML.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 24]
```diff
@@ -0,0 +1,208 @@
...
+    --no-enable-autoupgrade \
+    --release-channel=rapid \
+    --cluster-version=1.23
+```
+
+You will also need to configure `kubectl` to access the cluster:
+
+```shell
+gcloud container clusters get-credentials test-grpc
+```
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:96]: I'd definitely prefer an example that wasn't cloud-provider-specific. Nudging readers towards GKE isn't the right look.

It's OK to provide a _set_ of examples; for example, you could have examples for:
- GKE
- kind
- minikube

(we prefer to list in alphabetical order)

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 25]
```diff
@@ -0,0 +1,208 @@
...
+yourself with the `GRPCContainerProbe` feature gate enabled, there are many [tools available](/docs/tasks/tools/).
+
+Since the feature gate `GRPCContainerProbe` is enabled by default in 1.24,
+many vendors will have this functionality working out of the box.
+So you may just create an 1.24 cluster on platform of your choice. Some vendors
+allow to enable alpha features on 1.23 clusters.
+
+For example, at the moment of writing, you can spin up the test cluster on GKE for a quick test.
+Other vendors may also have similar capabilities, especially if you
+read this blog post long after kubernetes 1.24 released.
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:79]:
```suggestion
are reading this blog post long after the Kubernetes 1.24 release.
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 26]
```diff
@@ -0,0 +1,208 @@
...
+Built-in checks also cannot be configured to ignore certain types of errors
+(`grpc_health_probe` returns different exit codes for different errors),
+and cannot be "chained" to run the health check on multiple services in a single probe.
+
+But all these limitations are quite standard for gRPC and there are easy workarounds
+for those.
+
+## Try it for yourself
+
+### Getting cluster
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:67]:
```suggestion
### Cluster-level setup
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 27]
```diff
@@ -0,0 +1,208 @@
...
+
+You can try this feature today. To try the feature you can spin up the kubernetes cluster
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:69]:
```suggestion
You can try this feature today. To try native gRPC probes, you can spin up a Kubernetes cluster
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 28]
```diff
@@ -0,0 +1,208 @@
+---
+layout: blog
+title: "Try the new gRPC probes in 1.24: now in beta"
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:3]:
```suggestion
title: "Kubernetes 1.24: gRPC container probes in beta"
```
(and adjust the slug to match)

**Review by sftim — COMMENTED** [review]
A quick review. I rushed it a bit so I'm afraid I might have either missed something or misunderstood something.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 29]
```diff
@@ -0,0 +1,208 @@
...
+date: 2022-04-15
```
Comment (reylejano) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:4]: Publish date should be after 2022-03-05 since this blog is a feature blog for the 1.24 release

Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:4]:
```suggestion
date: 2022-05-05
```

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md (hunk 23 above)
Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:110]: did you mean to remove the apiVersion in code proposal?

Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:110]: I cannot find many examples at kubernetes.io where the snippets starts with `---`, but happy to apply if it's best

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md (hunk 24 above)
Comment (SergeyKanzhelev) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:96]: nudging wasn't an intention, I just wanted to provide something easier than kind to make people try it. Kind instructions will be a bit more involved that I'd prefer - changing feature gates will be needed.

if this is not ideal, I can remove that part completely only leaving the references to `kind`. Not sure if providing step-by-step instruction would be useful for anybody. People who will try it with kind, likely already know it. People who doesn't - don't need those.

wdyt?

**Comment by mickeyboxell** [comment]
@SergeyKanzhelev could we plan to have this ship on 10-MAY-2022?

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md (hunk 24 above)
Comment (sftim) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:96]: I'm fine with either option.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md (hunk 23 above)
Comment (sftim) [reply content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:110]: We used to have a bug where we couldn't process manifests that started `---` but (fortunately) that's gone now.

File: content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md [hunk 30]
```diff
@@ -0,0 +1,208 @@
+---
+layout: blog
+title: "Kubernetes 1.24: gRPC container probes in beta"
+date: 2022-05-05
```
Comment (sftim) [inline content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md:4]:
```suggestion
date: 2022-05-13
```
Please update the path to match; it doesn't affect the published article but it does help contributors.

**Review by sftim — CHANGES_REQUESTED** [review]

**Comment by SergeyKanzhelev** [comment]
@sftim updated and rebased

**Comment by mickeyboxell** [comment]
@SergeyKanzhelev can we change the title to 2022-05-13-grpc-probes-in-beta.md

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev can we change the title to 2022-05-13-grpc-probes-in-beta.md?

done

File: content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md [hunk 31]
```diff
@@ -0,0 +1,209 @@
...
+layout: blog
+title: "Kubernetes 1.24: gRPC container probes in beta"
+date: 2022-05-13
+slug: grpc-probes-now-in-beta
+---
+
+**Author**: Sergey Kanzhelev (Google)
+
+
+With Kubernetes 1.24 the gRPC probes functionality entered beta and is available by default.
```
Comment (divya-mohan0209) [inline content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md:11]: I'm just a bit curious here since starting K8s v1.24, beta APIs will not be available by default. Since this is not an existing beta API, I'm not sure if I'm misunderstanding something with this sentence.
```suggestion
With Kubernetes 1.24 the gRPC probes functionality entered beta and is available by default.
```

File: content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md [hunk 32]
```diff
@@ -0,0 +1,209 @@
...
+    command: ["/agnhost", "grpc-health-checking"]
+    ports:
+    - containerPort: 5000
+    - containerPort: 8080
+    readinessProbe:
+      grpc:
+        port: 5000
+```
+
+If the file called `test.yaml`, you can create the pod and check it's status.
```
Comment (divya-mohan0209) [inline content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md:128]: Do we intend to say - If the file called `test.yaml` is created?

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]
Initial set of suggestions/reviews. Hope you find it useful.

File: content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md [hunk 33]
```diff
@@ -0,0 +1,209 @@
...
+This new built-in gRPC health probing on Kubernetes makes implementing a health-check via gRPC
+much easier than the older approach that relied on using a separate `exec` probe. Read through
+the official
+[documentation](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe)
+to learn more and provide feedback before the feature will be promoted to GA.
+
+## Summary
+
+Kubernetes is a popular workload orchestration platform and we add features based on feedback and demand.
+Features like gRPC probes support is a minor improvement that will make life of many app developers
```
Comment (sftim) [inline content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md:208]:
```suggestion
A feature such as this one, adding support for gRPC probes, is a minor improvement that
will make the life of many app developers
```

File: content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md [hunk 34]
```diff
@@ -0,0 +1,209 @@
...
+
+This new built-in gRPC health probing on Kubernetes makes implementing a health-check via gRPC
+much easier than the older approach that relied on using a separate `exec` probe. Read through
+the official
+[documentation](/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-grpc-liveness-probe)
+to learn more and provide feedback before the feature will be promoted to GA.
+
+## Summary
+
+Kubernetes is a popular workload orchestration platform and we add features based on feedback and demand.
```
Comment (sftim) [inline content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md:207]:
```suggestion
Kubernetes is a popular workload orchestration platform;
contributors add features based on feedback and demand.
```
(“we” could read as Google which isn't wrong but also isn't the “we” that I think you meant to imply)

File: content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md [hunk 35]
```diff
@@ -0,0 +1,209 @@
...
+Features like gRPC probes support is a minor improvement that will make life of many app developers
+easier and apps more resilient. Try it today and give feedback, before the feature went into GA.
```
Comment (sftim) [inline content/en/blog/_posts/2022-05-13-grpc-probes-in-beta.md:209]: I'd reword this for both grammar and meaning. Some examples:
```suggestion
easier, and apps more resilient. Try it today and send in your feedback—I'd appreciate
hearing about people's real-world experience before the feature becomes generally available.
```
or
```suggestion
easier, and apps more resilient. Try it today and give feedback—I'd appreciate hearing
about people's real-world experience before the feature moves to GA.
```
or
```suggestion
easier, and apps more resilient. Try it today and give feedback—I'd appreciate hearing
about people's real-world experience before recommending that it graduates to a stable
feature.
```

**Review by sftim — COMMENTED** [review]
LGTM

A few tweaks would make the text even better.
/approve

**Comment by sftim** [comment]
Based on existing LGTM - see https://github.com/kubernetes/website/pull/32664#issuecomment-1100312660

/lgtm

@SergeyKanzhelev you are welcome - if you want to - to incorporate remaining feedback into a follow-up PR. We can merge that either before or after publication.
